Source/Client/Multiplayer.cs

```diff
@@ -193,16 +193,19 @@ private static void HandleCommandLine()
         {
             if (GenCommandLine.TryGetCommandLineArg("connect", out string ip))
             {
-                int port = MultiplayerServer.DefaultPort;
+                int port;
 
-                var split = ip.Split(':');
-                if (split.Length == 0)
+                MatchCollection matches = Regex.Matches(ip, "(.*)((?::))((?:[0-9]+))$");
+                if (matches.Count == 1) // Port is included, use regex
```

Comment:
This check won't handle plain IPv6 addresses properly.
For instance, `2001:0db8:85a3:0000:0000:8a2e:0370:7334` is a valid IP address that [will be matched](https://regex101.com/r/kjHMZF/1) as an IP:PORT pair.

Reply:
Thanks for the response.

That's correct. The regex I used is incomplete and needs expansion for IPv6 without port.

A simple check could be to count the number of `:` used in the string, as IPv4 will use 0 or 1, and IPv6 will use at least 7.

Reply:
IPv6 will actually use at least 3 colons, because of the short form, for example - `::1` is a valid IPv6 loopback address

Reply:
Interesting, didn't know that :)
though even in that case a similar rule would still work

Reply:
Maybe try using `[System.Net.IPAddress Parse (string ipString)](https://docs.microsoft.com/en-us/dotnet/api/system.net.ipaddress.parse?view=netframework-4.8)` instead of reinventing the wheel?